### PR TITLE
Added type hinting for weight_windows.py

### PR DESCRIPTION
**Pull request by christinacai123**

### CODE REVIEW

File: openmc/weight_windows.py [hunk 1]
```diff
@@ -18,7 +20,7 @@ class WeightWindows(IDManagerMixin):
     """Mesh-based weight windows
 
     This class enables you to specify weight window parameters that are used in
-    a simulation. Multiple sets of weight windows can be defined for different
+    a simulation. Multiple sets of weight windows can be defined for differen  t
```
Comment (shimwell) [inline openmc/weight_windows.py:23]:
```suggestion
    a simulation. Multiple sets of weight windows can be defined for different
```
looks like a some spaces got added here

**Review by christinacai123 — COMMENTED** [review]
fixed spacing on line 23

File: openmc/weight_windows.py [hunk 2]
```diff
@@ -103,16 +105,20 @@ class WeightWindows(IDManagerMixin):
...
-                 particle_type='neutron',
-                 survival_ratio=3,
-                 max_lower_bound_ratio=None,
-                 max_split=10,
-                 weight_cutoff=1.e-38,
-                 id=None):
+    def __init__(
+        self, 
+        mesh: MeshBase, 
+        lower_ww_bounds: np.ndarray,
```
Comment (paulromano) [inline openmc/weight_windows.py:111]:
```suggestion
        lower_ww_bounds: Iterable[float],
```

File: openmc/weight_windows.py [hunk 3]
```diff
@@ -103,16 +105,20 @@ class WeightWindows(IDManagerMixin):
...
+        upper_ww_bounds: Optional[np.ndarray] = None,
```
Comment (paulromano) [inline openmc/weight_windows.py:112]:
```suggestion
        upper_ww_bounds: Optional[Iterable[float]] = None,
```

File: openmc/weight_windows.py [hunk 4]
```diff
@@ -213,7 +219,7 @@ def energy_bounds(self) -> Iterable[Real]:
         return self._energy_bounds
 
     @energy_bounds.setter
-    def energy_bounds(self, bounds):
+    def energy_bounds(self, bounds: Iterable[Real]):
```
Comment (paulromano) [inline openmc/weight_windows.py:222]:
```suggestion
    def energy_bounds(self, bounds: Iterable[float]):
```

File: openmc/weight_windows.py [hunk 5]
```diff
@@ -247,7 +253,7 @@ def upper_ww_bounds(self) -> np.ndarray:
         return self._upper_ww_bounds
 
     @upper_ww_bounds.setter
-    def upper_ww_bounds(self, bounds):
+    def upper_ww_bounds(self, bounds: Iterable[Real]):
```
Comment (paulromano) [inline openmc/weight_windows.py:256]:
```suggestion
    def upper_ww_bounds(self, bounds: Iterable[float]):
```

File: openmc/weight_windows.py [hunk 6]
```diff
@@ -1,10 +1,12 @@
 from __future__ import annotations
 from collections.abc import Iterable
 from numbers import Real, Integral
-from typing import Iterable, List
+from typing import Iterable, List, Optional, Union
```
Comment (paulromano) [inline openmc/weight_windows.py:4]:
```suggestion
from typing import Iterable, List, Optional, Union, Dict
```

File: openmc/weight_windows.py [hunk 7]
```diff
@@ -398,7 +404,7 @@ def from_xml_element(cls, elem, root) -> WeightWindows:
         )
 
     @classmethod
-    def from_hdf5(cls, group, meshes) -> WeightWindows:
+    def from_hdf5(cls, group: h5py.Group, meshes: dict) -> WeightWindows:
```
Comment (paulromano) [inline openmc/weight_windows.py:407]:
```suggestion
    def from_hdf5(cls, group: h5py.Group, meshes: Dict[int, MeshBase]) -> WeightWindows:
```

File: openmc/weight_windows.py [hunk 8]
```diff
@@ -443,7 +449,7 @@ def from_hdf5(cls, group, meshes) -> WeightWindows:
         )
 
 
-def wwinp_to_wws(path) -> List[WeightWindows]:
+def wwinp_to_wws(path: Union[str, pathlib.Path]) -> List[WeightWindows]:
```
Comment (paulromano) [inline openmc/weight_windows.py:452]: In the checkvalue.py module, we have an alias for this called `PathLike` that is used elsewhere

**Review by paulromano — CHANGES_REQUESTED** [review]

**Review by christinacai123 — COMMENTED** [review]
added suggested changes

**Comment by christinacai123** [comment]
We failed the one of the checks and were wondering if you had any advice on how to fix this?

**Comment by christinacai123** [comment]
Unfortunately, I am still failing the same check. Could you try retriggering the CI again?

File: openmc/weight_windows.py [hunk 9]
```diff
@@ -1,14 +1,18 @@
 from __future__ import annotations
 from collections.abc import Iterable
 from numbers import Real, Integral
-from typing import Iterable, List
+import typing
+from typing import Iterable, List, Optional, Union, Dict
+import pathlib
```
Comment (paulromano) [inline openmc/weight_windows.py:6]:
```suggestion
import pathlib
import typing
from typing import Iterable, List, Optional, Union, Dict
```

**Review by paulromano — COMMENTED** [review]
Ah, it looks like you need to make a new commit on the branch for it to pick up updates from the `develop` branch in CI. I've just made a trivial suggestion that you can commit and hopefully that should do the trick.

**Review by paulromano — APPROVED** [review]

**Pull request merged**